packages/vm/src/instance.rs

```diff
@@ -147,17 +150,11 @@ where
         with_storage_from_context(self.wasmer_instance.context(), func)
     }
 
-    /// Copies all data described by the Region at the given pointer from Wasm to the caller.
-    pub(crate) fn memory(&self, region_ptr: u32) -> Vec<u8> {
-        read_region(self.wasmer_instance.context(), region_ptr)
-    }
-
-    /// Allocates memory in the instance and copies the given data into it.
-    /// Returns a pointer in the Wasm address space to the created Region object.
-    pub(crate) fn allocate(&mut self, data: &[u8]) -> Result<u32> {
+    /// Requests memory allocation by the instance and returns a pointer
+    /// in the Wasm address space to the created Region object.
+    pub(crate) fn allocate(&mut self, size: usize) -> Result<u32> {
```

Comment:
Why do you separate allocate and the write_region? 
(Not that it is wrong, just curious)

Reply:
Originally to improve testability. Anything lower level than `Instance` does not have tests. The additional expliciteness allows testing one thing at a time.

Then there is symmetry as well. The term "allocate" was used differently here than in the rest of the stack, as it was implemented as allocate+write.

This is especially useful now that we can allocate memory without every writing to it.